stock_delivery_notification/data/mail_template_data.xml

```diff
@@ -24,7 +24,13 @@ This is to inform you that your order <a href="${object.web_url}"><strong>${obje
 <p>
 <u>Delivery Details</u><br/>
 Date Shipped: ${object.date_done_ctx}<br/>
-Carrier: ${object.carr_id.name.replace('Stamps.com', 'USPS') or 'N/A'}<br/>
+Carrier:
+% if object.carr_id.name:
+    ${object.carr_id.name.replace('Stamps.com', 'USPS')}
+% else:
+    N/A
+% endif
+<br/>
```

Comment:
Can you check if `Carrier: ${object.carr_id.name.replace('Stamps.com', 'USPS') if object.carr_id else 'N/A'}<br/>` would do? If not, we can go with your code.